common/solver/bicg_kernels.cpp

```diff
@@ -63,57 +60,60 @@ void initialize(std::shared_ptr<const OmpExecutor> exec,
                 matrix::Dense<ValueType> *q2,
                 Array<stopping_status> *stop_status)
 {
-#pragma omp parallel for
-    for (size_type j = 0; j < b->get_size()[1]; ++j) {
-        rho->at(j) = zero<ValueType>();
-        prev_rho->at(j) = one<ValueType>();
-        stop_status->get_data()[j].reset();
-    }
-#pragma omp parallel for
-    for (size_type i = 0; i < b->get_size()[0]; ++i) {
-        for (size_type j = 0; j < b->get_size()[1]; ++j) {
-            r->at(i, j) = b->at(i, j);
-            r2->at(i, j) = b->at(i, j);
-            z->at(i, j) = p->at(i, j) = q->at(i, j) = zero<ValueType>();
-            z2->at(i, j) = p2->at(i, j) = q2->at(i, j) = zero<ValueType>();
-        }
-    }
+    run_kernel(
+        exec,
+        [] GKO_KERNEL(auto row, auto col, auto b, auto r, auto z, auto p,
+                      auto q, auto prev_rho, auto rho, auto r2, auto z2,
+                      auto p2, auto q2, auto stop) {
+            if (row == 0) {
+                rho[col] = zero(rho[col]);
+                prev_rho[col] = one(prev_rho[col]);
+                stop[col].reset();
+            }
+            r(row, col) = b(row, col);
+            r2(row, col) = b(row, col);
+            z(row, col) = zero(z(row, col));
+            p(row, col) = zero(p(row, col));
+            q(row, col) = zero(q(row, col));
+            z2(row, col) = zero(z2(row, col));
+            p2(row, col) = zero(p2(row, col));
+            q2(row, col) = zero(q2(row, col));
+        },
+        p->get_size(), b, compact(r), compact(z), compact(p), compact(q),
```

Comment:
I think we would like to have new create_with_same_size_of(b), so the others except for b have the same stride

Reply:
Yes, the implementation currently does not respect non-default strides in `b` yet. I would personally prefer changing the behavior of `create_with_same_config` to not copy the stride, since this would prevent any issues related to users overriding Dense::create_with_same_config (considering MFEM especially)